cura/Settings/SettingOverrideDecorator.py

```diff
@@ -90,10 +90,13 @@ def getActiveExtruderPosition(self):
     def isNonPrintingMesh(self):
         return self._is_non_printing_mesh
 
+    def evaluateIsNonPrintingMesh(self):
+        return any(bool(self._stack.getProperty(setting, "value")) for setting in self._non_printing_mesh_settings)
+
     def _onSettingChanged(self, instance, property_name): # Reminder: 'property' is a built-in function
         # Trigger slice/need slicing if the value has changed.
-        if property_name == "value":
-            self._is_non_printing_mesh = any(bool(self._stack.getProperty(setting, "value")) for setting in self._non_printing_mesh_settings)
+        if property_name == "value" and instance in self._non_printing_mesh_settings:
```

Comment:
This is quite interesting, it works fine for general settings. But if you set setting per model setting, I expect cura starts slicing. Can you check it?